langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/RequireType.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.langlib.value;
+
+import org.ballerinalang.jvm.TypeChecker;
+import org.ballerinalang.jvm.scheduling.Strand;
+import org.ballerinalang.jvm.types.BType;
+import org.ballerinalang.jvm.values.ErrorValue;
+import org.ballerinalang.jvm.values.TypedescValue;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+
+import static org.ballerinalang.util.BLangCompilerConstants.VALUE_VERSION;
+
+/**
+ * Extern function lang.values:requireType.
+ *
+ * @since Swan Lake
+ */
+@BallerinaFunction(
+        orgName = "ballerina", packageName = "lang.value", version = VALUE_VERSION,
+        functionName = "requireType",
+        args = {@Argument(name = "v", type = TypeKind.UNION), @Argument(name = "type", type = TypeKind.TYPEDESC)},
+        returnType = {@ReturnType(type = TypeKind.UNION), @ReturnType(type = TypeKind.ERROR)},
+        isPublic = false
+)
+
+public class RequireType {
+    public static Object requireType(Strand strand, Object value, TypedescValue type) {
+        if (value instanceof ErrorValue) {
```

Comment:
We can use `org.ballerinalang.jvm.TypeChecker#getType` and then check the type tag instead of the `instanceof` check.

Reply:
Changed.